src/include/common/types/types.h

```diff
@@ -241,6 +241,8 @@ class ExtraTypeInfo;
 class StructField;
 class StructTypeInfo;
 
+enum class TypeInfo : uint8_t { INTERNAL = 0, UDT = 1 };
```

Comment:
TypeInfo is conflicting with `ExtraTypeInfo`. `TypeType` also sounds weird. Let's maybe call this `TypeClass` or `TypeCatagory`